src/shared/components/plots/MultipleCategoryBarPlotUtils.ts

```diff
@@ -98,6 +98,71 @@ export function sortDataByCategory<D>(
         }
     });
 }
+export function getSortedMajorCategories(
+    data: IMultipleCategoryBarPlotData[],
+    sortByOption: string | undefined
+): string[] {
+    if (sortByOption === 'SortByTotalSum') {
+        const majorCategoryCounts: any = {};
+
+        data.forEach(item => {
+            item.counts.forEach(countItem => {
+                const { majorCategory, count } = countItem;
+                if (!majorCategoryCounts[majorCategory]) {
+                    majorCategoryCounts[majorCategory] = 0;
+                }
+                majorCategoryCounts[majorCategory] += count;
+            });
+        });
+
+        return Object.keys(majorCategoryCounts).sort(
+            (a, b) => majorCategoryCounts[b] - majorCategoryCounts[a]
+        );
+    } else if (sortByOption !== '' && sortByOption !== 'alphabetically') {
+        const sortedEntityData = data.find(
+            item => item.minorCategory === sortByOption
+        );
+        if (sortedEntityData) {
+            sortedEntityData.counts.sort((a, b) => b.count - a.count);
+
+            return sortedEntityData.counts.map(item => item.majorCategory);
+        }
+    }
+
+    return [];
+}
+export function sortDataByOption(
+    data: IMultipleCategoryBarPlotData[],
+    sortByOption: string
+): IMultipleCategoryBarPlotData[] {
+    const sortedMajorCategories = getSortedMajorCategories(data, sortByOption);
+
+    if (sortByOption === 'SortByTotalSum' || sortedMajorCategories.length > 0) {
+        const reorderCounts = (counts: any) => {
```

Comment:
avoid 'any'

Reply:
Yes, I have defined the types across all the changes.